doc/Design.md

```diff
@@ -147,19 +147,11 @@ Notes:
 
 ## Address Representations
 
-There are three different representations of addresses in Macaw, which often leads to confusion.  This section will attempt to describe them and their intended use cases. The address types are parameterized by the number of bits in the address. Each address type has a type alias prefixed with `Arch` that is instead parameterized by architecture, with the width parameter filled in according to the declared width of the architecture.
+There are four different representations of addresses in Macaw, which often leads to confusion. `MemAddr`, `MemWord`, and `MemSegmentOff` are used in conjunction with the *pre-loader* view of memory. They are disambiguated in the module-level documentation for `Data.Macaw.Memory`.
```

Comment:
If you want to be pedantic, you could argue that there are actually _five_ such types if you include `MemInt`... perhaps we should change "four" to "several"?

Reply:
I would argue that `MemInt` is not a representation of an address, as it is signed. Will change to "several".